warn/multifile.go

```diff
@@ -0,0 +1,48 @@
+package warn
+
+import (
+	"github.com/bazelbuild/buildtools/build"
+)
+
+// FileReader is a class that can read an arbitrary Starlark file
+// from the repository and cache the results.
+type FileReader struct {
+	cache    map[string]*build.File
+	readFile func(string) ([]byte, error)
+}
+
+// Init initializes a FileReader instance with an instance of function
+// that can read an arbitrary file in the repository using a path relative
+// to the workspace root.
+func (fr *FileReader) Init(readFile func(string) ([]byte, error)) {
+	fr.cache = make(map[string]*build.File)
+	fr.readFile = readFile
+}
+
+// retrieveFile reads a Starlark file using only the readFile method
+// (without using the cache).
+func (fr *FileReader) retrieveFile(filename string) *build.File {
+	contents, err := fr.readFile(filename)
+	if err != nil {
+		return nil
```

Comment:
log this and the below error?

Reply:
I tried to, but log messages appear in the stderr where the error messages normally appear, I thought it would be confusing for the users.